crates/polars-utils/src/idx_vec.rs

```diff
@@ -99,8 +99,10 @@ impl<T> UnitVec<T> {
         }
     }
 
+    /// # Panics
+    /// Panics if `new_cap < self.len` or `new_cap == 0`
     fn realloc(&mut self, new_cap: usize) {
-        assert!(new_cap >= self.len);
+        assert!(new_cap >= self.len.max(1));
```

Comment:
No, it needs to be *bigger* than 1, not bigger or equal. Capacity == 1 is a cursed special value. Doc comment also needs update.